app/helpers/view_components/widgets/exercise.rb

```diff
@@ -28,33 +29,15 @@ def to_s
         end
       end
 
-      # TODO: We want to have a solutions list that
-      # pre-fetches all the relevant solutions for a
-      # user, which we then pass into this. This whole
-      # method should then be replaces and the solution
-      # should be passed in directly.
-      memoize
-      def solution
-        # TODO: This will break once devise is added
-        Solution.for(User.first, exercise)
-      end
-
```

Comment:
This is an example of what all this new code is fixing. It removes this `n+1`, which is deeply nested in a view component, and ensures we only calculate this data once across a whole request (at most!)